src/ds/d500/d500-fw-update-device.cpp

```diff
@@ -38,6 +38,13 @@ ds_d500_update_device::ds_d500_update_device( std::shared_ptr< const device_info
     void ds_d500_update_device::update(const void* fw_image, int fw_image_size, rs2_update_progress_callback_sptr update_progress_callback) const
     {
         update_device::update( fw_image, fw_image_size, update_progress_callback );
+
+        if (_wait_instead_of_sampling_manifest_reset)  // when dfu monitoring is not enabled by FW
+        {
+            LOG_DEBUG("Waiting for the FW to be burnt");
+            static constexpr int D500_FW_DFU_TIME = 120; // [sec]
+            report_progress_and_wait_for_fw_burn(update_progress_callback, D500_FW_DFU_TIME);
+        }
     }
 
     bool ds_d500_update_device::wait_for_manifest_completion(std::shared_ptr<platform::usb_messenger> messenger, const rs2_dfu_state state,
```

Comment:
Since this is a new function maybe we can add another parameter we write on that will say `support_progress_quering` or "support_manifest` or something like that instead of writing on a new member?

Reply:
I cannot see a better way. Since you gave me the choice, this will remain.